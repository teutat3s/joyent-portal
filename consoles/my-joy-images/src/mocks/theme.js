import React from 'react';
import { ThemeProvider } from 'styled-components';
import {
  theme,
  RootContainer,
  PageContainer,
  ViewContainer
} from 'joyent-ui-toolkit';

export default ({ children, ss }) => (
  <ThemeProvider theme={theme}>
    {ss ? (
      <RootContainer>
        <PageContainer>
          <ViewContainer>{children}</ViewContainer>
        </PageContainer>
      </RootContainer>
    ) : (
      children
    )}
  </ThemeProvider>
);
