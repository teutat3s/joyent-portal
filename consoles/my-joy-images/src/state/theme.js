import { theme } from 'joyent-ui-toolkit';

const font = theme.font.href({
  namespace: 'images'
});

const monoFont = theme.monoFont.href({
  namespace: 'images'
});

export default {
  ...theme,
  font: {
    ...theme.font,
    href: () => font
  },
  monoFont: {
    ...theme.monoFont,
    href: () => monoFont
  }
};
