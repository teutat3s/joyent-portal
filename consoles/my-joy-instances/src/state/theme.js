import { theme } from 'joyent-ui-toolkit';

const font = theme.font.href({
  namespace: 'instances'
});

const monoFont = theme.monoFont.href({
  namespace: 'instances'
});

export default {
  ...theme,
  font: {
    ...theme.font,
    href: () => font
  },
  monoFont: {
    ...theme.monoFont,
    href: () => monoFont
  }
};
