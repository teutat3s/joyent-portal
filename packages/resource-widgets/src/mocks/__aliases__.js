module.exports = {
  '^redux-form$': '<rootDir>/src/mocks/redux-form'
};
