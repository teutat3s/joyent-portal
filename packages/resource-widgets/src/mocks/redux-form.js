import React from 'react';

export const Field = ({ component = 'input', children, ...rest }) =>
  React.createElement(component, rest, children);
