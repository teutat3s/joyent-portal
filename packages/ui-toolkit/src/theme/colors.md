```js noeditor
const React = require('react');
const { default: Colors } = require('./colors');

<Colors />;
```
