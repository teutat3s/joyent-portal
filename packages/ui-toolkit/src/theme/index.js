import remcalc from 'remcalc';
import { canUseDOM } from 'exenv';

const globals = (() => {
  if (!canUseDOM) {
    return {};
  }

  return {
    protocol: window.location.protocol,
    host: window.location.host
  };
})();

const flexboxgrid = {
  gridSize: 12, // rem
  gutterWidth: 1.25, // rem
  outerMargin: 1.875, // rem
  mediaQuery: 'only screen',
  container: {
    sm: 46, // rem
    md: 56, // rem
    lg: 59 // rem
  },
  breakpoints: {
    xs: 0, // em
    sm: 48, // em
    md: 64, // em
    lg: 75 // em
  }
};

const spacing = {
  0: remcalc(0),
  0.5: remcalc(3),
  1: remcalc(6),
  2: remcalc(12),
  3: remcalc(18),
  4: remcalc(30),
  5: remcalc(48)
};

/*
* There should be no string value to keys outside the `base` object.
* If a new colors needs to be used, check it doesn't already exist, or
* anything that is similar, and if it doesn't, add a key-value and reference from
* there. Lets try and keep different color variations down ot a minimum.
*
---
Color Object example
const color_name = {
  color_name: '#FFFFFF',
  color_name_style1: '#FFF000',
  color_name_style2: '#FFF333',
};
*/

const primary = {
  primary: 'rgb(59, 70, 204)',
  primaryHover: 'rgb(72, 83, 217)',
  primaryActive: 'rgb(45, 56, 132)'
};

// TOPOLOGY
const secondary = {
  secondary: 'rgb(70, 70, 70)',
  secondaryHover: 'rgb(53, 53, 53)',
  secondaryActive: 'rgb(45, 45, 45)'
};

const white = {
  white: 'rgb(255, 255, 255)',
  whiteHover: 'rgb(247, 247, 247)',
  whiteActive: 'rgb(230, 230, 230)'
};

const grey = {
  grey: 'rgb(216, 216, 216)',
  greyTransparent: 'rgba(73, 73, 73, 0.8)',
  greyLight: 'rgb(189, 189, 189)',
  greyDark: 'rgb(151, 151, 151)',
  greyDarker: 'rgb(052, 052, 052)'
};

const green = {
  green: 'rgb(0, 152, 88)',
  greenDark: 'rgb(0, 129, 56)'
};

const orange = {
  orange: 'rgb(227, 130, 0)',
  orangeDark: 'rgb(203, 116, 0)'
};

const red = {
  red: 'rgb(210, 67, 58)',
  redDark: 'rgb(205, 37, 27)'
};

/** ********************************** BASE *********************************** */

export const base = {
  ...primary,
  ...secondary,
  ...white,
  ...red,
  ...orange,
  ...green,
  ...grey,
  text: 'rgb(73, 73, 73)',
  textDisabled: 'rgba(73, 73, 73, 0.5)',
  placeholder: 'rgb(99, 99, 99)',
  disabled: 'rgb(250, 250, 250)', // used
  background: 'rgb(250, 250, 250)' // used
};

/** ********************************** HEADER ********************************** */

const brandBackground = 'rgb(30, 49, 59)';

/** ********************************** FONTS ********************************** */

export const font = {
  semibold: base.secondary,
  regular: base.text,
  abbrBorderColor: base.secondary,
  textMuted: base.secondary,
  family: '"Libre Franklin"',
  families:
    '"Libre Franklin", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, Helvetica, sans-serif',
  href: ({ protocol, host, namespace } = {}) => {
    const _protocol = protocol || globals.protocol;
    const _host = host || globals.host;

    return `${_protocol}//${_host}/${namespace}/static/css/libre-franklin.css`;
  },
  weight: {
    bold: 700,
    semibold: 600,
    medium: 500,
    normal: 400
  }
};

export const monoFont = {
  semibold: base.secondary,
  regular: base.text,
  abbrBorderColor: base.secondary,
  textMuted: base.secondary,
  family: '"Roboto Mono"',
  families: '"Roboto Mono", monospace',
  href: ({ protocol, host, namespace } = {}) => {
    const _protocol = protocol || globals.protocol;
    const _host = host || globals.host;

    return `${_protocol}//${_host}/${namespace}/static/css/roboto-mono.css`;
  },
  weight: {
    bold: 700,
    normal: 400
  }
};

/** ******************************** INACTIVE ********************************* */

export const inactive = {
  default: base.disabled,
  border: base.grey,
  text: base.grey
};

/** ***************************** NOTIFICATIONS ******************************* */

export const notifications = {
  alert: base.red,
  confirmation: base.green,
  warning: base.orange
};

/** ********************************* FORMS *********************************** */

export const inputError = base.red;
export const inputWarning = base.orange;

/** ******************************** METRICS ********************************* */

/* export const miniBackground = '#F3F4F9';
export const separator = '#D9DEF3'; */

/** ******************************** TOPOLOGY ********************************* */

export const topologyBackground = base.secondaryActive;

export const transition = 'all 200ms ease-out';

export const borderRadius = remcalc(4);

export const shadows = {
  bottomShadow: `0 ${remcalc(2)} 0 0 rgba(0, 0, 0, 0.05)`,
  bottomShadowDarker: `0 ${remcalc(2)} 0 0 rgba(0, 0, 0, 0.1)`
};

export default {
  ...base,
  flexboxgrid,
  spacing,
  font,
  monoFont,
  inactive,
  notifications,
  inputError,
  inputWarning,
  topologyBackground,
  brandBackground,
  transition,
  borderRadius,
  shadows,
  transparent: 'transparent'
};
