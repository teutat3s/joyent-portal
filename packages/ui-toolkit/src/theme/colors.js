import React, { Fragment } from 'react';
import styled from 'styled-components';
import remcalc from 'remcalc';
import { P, H3 } from '../text';
import is from 'styled-is';
import theme from './';
import Flex from 'styled-flex-component';
import { Margin } from 'styled-components-spacing';
import copy from 'clipboard-copy';
import { Clipboard } from '../icons';

// Function to convert hex format to a rgb color
function rgb2hex(rgb) {
  rgb = rgb.match(
    /^rgba?[\s+]?\([\s+]?(\d+)[\s+]?,[\s+]?(\d+)[\s+]?,[\s+]?(\d+)[\s+]?/i
  );
  return rgb && rgb.length === 4
    ? '#' +
        ('0' + parseInt(rgb[1], 10).toString(16)).slice(-2) +
        ('0' + parseInt(rgb[2], 10).toString(16)).slice(-2) +
        ('0' + parseInt(rgb[3], 10).toString(16)).slice(-2)
    : '';
}

const parseRGB = c => {
  const color = c.split('(');
  const rgb = color[1].split(')')[0];

  return `${color[0].toUpperCase()}: ${rgb}`;
};

const Box = styled.div`
  width: ${remcalc(187)};
  height: ${remcalc(126)};
  background: ${props => props.hex};
  padding: ${remcalc(18)} ${remcalc(26)};
  box-sizing: border-box;
`;

const ClipboardIconActionable = styled(Clipboard)`
  cursor: pointer;

  path {
    fill: ${props => props.theme.white};
  }

  ${is('dark')`
    path { fill: ${props => props.theme.text}; }
  `};
`;

const Paragraph = styled(P)`
  font-size: ${remcalc(13)};
  font-weight: bold;
  color: ${props => props.theme.white};
  -webkit-text-fill-color: currentcolor;
  width: ${remcalc(193)};
  text-align: left;

  ${is('dark')`
    color: ${props => props.theme.text};
    -webkit-text-fill-color: currentcolor;
  `};
`;

const Code = styled.code`
  font-size: ${remcalc(13)};
  color: ${props => props.theme.white};

  ${is('dark')`
    color: ${props => props.theme.text};
    -webkit-text-fill-color: currentcolor;
  `};
`;

const List = styled.ul`
  list-style: none;
  display: flex;
  flex-direction: column;
  padding: 0;
  height: ${remcalc(48)};

  li {
    padding: 0;
  }
`;

const Color = ({ name, color, dark, bottom, right }) => (
  <Box hex={color} bottom={bottom} right={right}>
    <List>
      <li>
        <Paragraph dark={dark}>{name}</Paragraph>
      </li>
      <li>
        <Flex alignCenter>
          <Code dark={dark}>{rgb2hex(color).toUpperCase()}</Code>{' '}
          <ClipboardIconActionable
            dark={dark}
            onClick={() => copy(rgb2hex(color).toUpperCase())}
          />
        </Flex>
      </li>
      <li>
        <Code dark={dark}>{parseRGB(color)}</Code>
      </li>
    </List>
  </Box>
);

export default () => (
  <Fragment>
    <div>
      <H3>Action Colors</H3>
      <P>
        This palette contains Triton’s ‘action and status’ colors. They aim to
        communicate that a component is interactable and has a purpose. They
        also act as status colors to alert users on the condition and nature of
        components.
      </P>
    </div>
    <Margin style={{ display: 'inline-block' }} top={4} bottom={5} right={5}>
      <Color key="primaryHover" name="Blue 1" color={theme.primary} />
      <Color key="primaryActive" name="Blue 2" color={theme.primaryActive} />
    </Margin>
    <Margin style={{ display: 'inline-block' }} top={4} bottom={5}>
      <Color key="joyent1" name="Joyent 1" color="rgb(44, 72, 89)" />
      <Color key="joyent2" name="Joyent 2" color="rgb(27, 50, 64)" />
    </Margin>
    <div>
      <Margin right={5} style={{ display: 'inline-block' }} bottom={2}>
        <Color key="green" name="Green 1" color={theme.green} />
        <Color key="greenDark" name="Green 2" color={theme.greenDark} />
      </Margin>
      <Margin bottom={2} right={5} style={{ display: 'inline-block' }}>
        <Color key="orange" name="Orange 1" color={theme.orange} />
        <Color key="orangeDark" name="Orange 2" color={theme.orangeDark} />
      </Margin>
      <Margin bottom={5} right={5} style={{ display: 'inline-block' }}>
        <Color key="red" name="Red 1" color={theme.red} />
        <Color key="redDark" name="Red 2" color={theme.redDark} />
      </Margin>
    </div>
    <H3>Greys</H3>
    <P>
      Greys give Triton a sense of depth and offer contrast between potentially
      similar components. They allow us to make certain components look
      clickable, whilst making others appear disabled or static.
    </P>
    <Margin top={4}>
      <Flex wrap>
        <Color
          bottom
          right
          dark
          key="disabled"
          name="Grey 1 - Background"
          color={theme.disabled}
        />
        <Color
          bottom
          right
          dark
          key="whiteActive"
          name="Grey 2 - Faded"
          color={theme.whiteActive}
        />
        <Color
          bottom
          dark
          key="grey"
          name="Grey 3  - Dividers"
          color={theme.grey}
        />
        <Color
          bottom
          right
          key="greyDark"
          name="Grey 4  - Disabled"
          color={theme.greyDark}
        />
        <Color
          bottom
          right
          key="text"
          name="Grey 5  - Text"
          color={theme.text}
        />
        <Color bottom key="greyDarker" name="Grey 6" color={theme.greyDarker} />
      </Flex>
    </Margin>
  </Fragment>
);
