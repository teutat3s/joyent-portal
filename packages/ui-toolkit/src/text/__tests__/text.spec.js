import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import remcalc from 'remcalc';

import { P, H1, H2, H3, H4, H5, H6, Small, Sup } from '../';
import { Theme } from '../../mocks';
import theme from '../../theme';

describe('Button', () => {
  test('Paragraph', () => {
    const tree = renderer
      .create(
        <Theme>
          <P>Inspire the lazy</P>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.text.replace(/ /g, ''));
    expect(tree).toHaveStyleRule('font-size', remcalc(15));
  });

  test('Small', () => {
    const tree = renderer
      .create(
        <Theme>
          <Small>Inspire the lazy</Small>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.text.replace(/ /g, ''));
    expect(tree).toHaveStyleRule('font-size', remcalc(13));
  });

  test('H1', () => {
    const tree = renderer
      .create(
        <Theme>
          <H1>Inspire the lazy</H1>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.text.replace(/ /g, ''));
    expect(tree).toHaveStyleRule('font-size', remcalc(36));
  });

  test('H2', () => {
    const tree = renderer
      .create(
        <Theme>
          <H2>Inspire the lazy</H2>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.text.replace(/ /g, ''));
    expect(tree).toHaveStyleRule('font-size', remcalc(24));
  });

  test('H3', () => {
    const tree = renderer
      .create(
        <Theme>
          <H3>Inspire the lazy</H3>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.text.replace(/ /g, ''));
    expect(tree).toHaveStyleRule('font-size', remcalc(21));
  });

  test('H4', () => {
    const tree = renderer
      .create(
        <Theme>
          <H4>Inspire the lazy</H4>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.text.replace(/ /g, ''));
    expect(tree).toHaveStyleRule('font-size', remcalc(15));
    expect(tree).toHaveStyleRule(
      'font-weight',
      `${theme.font.weight.normal}`
    );
  });

  test('H5', () => {
    const tree = renderer
      .create(
        <Theme>
          <H5>Inspire the lazy</H5>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.text.replace(/ /g, ''));
    expect(tree).toHaveStyleRule('font-size', remcalc(15));
  });

  test('H6', () => {
    const tree = renderer
      .create(
        <Theme>
          <H6>Inspire the lazy</H6>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.text.replace(/ /g, ''));
    expect(tree).toHaveStyleRule('font-size', remcalc(13));
  });

  test('Sup', () => {
    const tree = renderer
      .create(
        <Theme>
          <Sup>Inspire the lazy</Sup>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.primary.replace(/ /g, ''));
    expect(tree).toHaveStyleRule('font-size', remcalc(8));
  });

  test('Sup badge', () => {
    const tree = renderer
      .create(
        <Theme>
          <Sup badge>Inspire the lazy</Sup>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.white.replace(/ /g, ''));
    expect(tree).toHaveStyleRule('font-size', remcalc(8));
  });
});
