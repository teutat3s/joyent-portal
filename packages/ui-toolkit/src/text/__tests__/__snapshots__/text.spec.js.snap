// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button H1 1`] = `
.c1 {
  font-size: 2rem;
  margin: 0.625rem 0;
}

.c0 {
  margin: 0;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  color: rgb(73,73,73);
  font-size: 2.25rem;
  line-height: 2.8125rem;
  font-style: normal;
  font-stretch: normal;
  font-weight: normal;
  margin: 0;
}

<h1
  className="c0 c1"
>
  Inspire the lazy
</h1>
`;

exports[`Button H2 1`] = `
.c0 {
  margin: 0;
  color: rgb(73,73,73);
  font-weight: normal;
  line-height: 1.875rem;
  font-size: 1.5rem;
}

<h2
  className="c0"
>
  Inspire the lazy
</h2>
`;

exports[`Button H3 1`] = `
.c0 {
  margin: 0;
  color: rgb(73,73,73);
  font-weight: normal;
  line-height: 1.625rem;
  font-size: 1.3125rem;
}

<h3
  className="c0"
>
  Inspire the lazy
</h3>
`;

exports[`Button H4 1`] = `
.c0 {
  margin: 0;
  color: rgb(73,73,73);
  font-weight: 400;
  line-height: 1.5rem;
  font-size: 0.9375rem;
}

<h4
  className="c0"
>
  Inspire the lazy
</h4>
`;

exports[`Button H5 1`] = `
.c0 {
  margin: 0;
  color: rgb(73,73,73);
  font-weight: 400;
  line-height: 1.5rem;
  font-size: 0.9375rem;
}

<h4
  className="c0"
>
  Inspire the lazy
</h4>
`;

exports[`Button H6 1`] = `
.c0 {
  margin: 0;
  color: rgb(73,73,73);
  font-weight: 400;
  line-height: 1.125rem;
  font-size: 0.8125rem;
}

<h6
  className="c0"
>
  Inspire the lazy
</h6>
`;

exports[`Button Paragraph 1`] = `
.c0 {
  color: rgb(73,73,73);
  margin: 0;
  line-height: 1.5rem;
  font-size: 0.9375rem;
  margin: 0;
}

<p
  className="c0"
>
  Inspire the lazy
</p>
`;

exports[`Button Small 1`] = `
.c1 {
  font-size: 80%;
}

.c0 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  color: rgb(73,73,73);
  line-height: 1.125rem;
  font-size: 0.8125rem;
  padding-bottom: 0.75rem;
}

<small
  className="c0 c1"
>
  Inspire the lazy
</small>
`;

exports[`Button Sup 1`] = `
.c0 {
  position: absolute;
  font-weight: 600;
  line-height: normal;
  font-size: 0.5rem;
  color: rgb(59,70,204);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<sup
  className="c0"
>
  Inspire the lazy
</sup>
`;

exports[`Button Sup badge 1`] = `
.c0 {
  position: absolute;
  font-weight: 600;
  line-height: normal;
  font-size: 0.5rem;
  color: rgb(59,70,204);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.1875rem;
  padding: 0.125rem 0.1875rem;
  background: rgb(59,70,204);
  color: rgb(255,255,255);
}

<sup
  className="c0"
>
  Inspire the lazy
</sup>
`;
